monitor/flowgen/symbiotic/symbiotic.go

```diff
@@ -0,0 +1,148 @@
+package symbiotic
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/omni-network/omni/contracts/bindings"
+	"github.com/omni-network/omni/lib/contracts"
+	"github.com/omni-network/omni/lib/contracts/solvernet"
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/ethclient/ethbackend"
+	"github.com/omni-network/omni/lib/netconf"
+	"github.com/omni-network/omni/monitor/flowgen/types"
+	"github.com/omni-network/omni/monitor/flowgen/util"
+	solver "github.com/omni-network/omni/solver/app"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+
+	_ "embed"
+)
+
+//go:embed symbiotic_abi.json
+var abiJSON []byte
+
+func NewJob(
+	ctx context.Context,
+	backends ethbackend.Backends,
+	networkID netconf.ID,
+	owner common.Address,
+	amount *big.Int,
+) (types.Job, error) {
+	addrs, err := contracts.GetAddresses(ctx, networkID)
+	if err != nil {
+		return types.Job{}, errors.New("contract addresses")
+	}
+
+	conf, ok := config[networkID]
+	if !ok {
+		return types.Job{}, errors.Wrap(err, "flow config missing")
+	}
+
+	backend, err := backends.Backend(conf.srcChain)
+	if err != nil {
+		return types.Job{}, errors.Wrap(err, "get backend")
+	}
+
+	srcChainTkn, ok := solver.AllTokens().FindBySymbol(conf.srcChain, conf.depositToken.Symbol)
+	if !ok {
+		return types.Job{}, errors.Wrap(err, "src token not found")
+	}
+
+	dstChainTkn, ok := solver.AllTokens().FindBySymbol(conf.dstChain, conf.expenseToken.Symbol)
```

Comment:
please add a TODO to solver/app.Tokens to move it to a separate package. It isn't good for other services (like monitor) to import the `app` package from another service. That is a very large dependency.